account/src/account/info.rs

```diff
@@ -0,0 +1,207 @@
+                address
+            };
+
+            let code_metadata = if let Some(raw_code_metadata) = raw_info.code_metadata {
+                Some(decode_code_meta_data(raw_code_metadata)?)
+            } else {
+                None
+            };
+
+            Ok(AccountInfo {
+                nonce: raw_info.nonce,
+                balance,
+                username: raw_info.username,
+                code: raw_info.code,
+                code_hash: raw_info.code_hash,
+                root_hash: raw_info.root_hash,
+                code_metadata: code_metadata,
+                developer_reward: developer_reward,
+                owner_address,
+            })
+        }
+    ).await
+}
+
+pub fn decode_code_meta_data(encoded: String) -> Result<CodeMetadata, AccountError> {
+    let decoded_bytes = base64::engine::general_purpose::STANDARD.decode(encoded.clone()).or(Err(AccountError::CannotDecodeCodeMetadata { metadata: encoded.clone() }))?;
+    if decoded_bytes.len() != 2 {
+        return Err(AccountError::CannotDecodeCodeMetadata { metadata: encoded });
+    }
+
+    let byte_array: [u8; 2] = decoded_bytes.as_slice().try_into().or(Err(AccountError::CannotDecodeCodeMetadata { metadata: encoded }))?;
+    Ok(CodeMetadata::from(byte_array))
+}
+
+#[cfg(test)]
+mod tests {
+    use num_bigint::BigUint;
+    use novax::caching::CachingNone;
+    use novax_data::Address;
+    use crate::account::info::fetch_account_info_for_address;
+    use crate::mock::request::MockClient;
+
+    #[tokio::test]
+    pub async fn test_with_valid_sc_address() {
+        let result = fetch_account_info_for_address(&MockClient::new(), &"erd1qqqqqqqqqqqqqpgqr7een4m5z44frr3k35yjdjcrfe6703cwdl3s3wkddz".into(), &CachingNone).await.unwrap();
+
+        assert_eq!(result.nonce, 0);
+        assert_eq!(result.balance, BigUint::from(0u64));
+        assert_eq!(result.username, "".to_string());
+        assert_eq!(result.code,"fakecodestring".to_string(),);
+        assert_eq!(result.code_hash, Some("gVgRRf6HhmTGlxziasAFoCgBlP7/DH0i9IhTbj7lsxA=".to_string()));
+        assert_eq!(result.root_hash, "A3RZ7aYh4NzkunNL+fu09ggnItEeC7SuPWJDfIHmAcI=".to_string());
+        if let Some(code_metadata) = result.code_metadata {
+            assert!(code_metadata.is_upgradeable());
+            assert!(code_metadata.is_readable());
+            assert!(!code_metadata.is_payable());
+            assert!(!code_metadata.is_payable_by_sc());
+        }
+        assert_eq!(result.developer_reward, BigUint::from(2288888045322000000u64));
+        assert_eq!(result.owner_address, Address::from_bech32_string("erd1kj7l40rmklhp06treukh8c2merl2h78v2939wyxwc5000t25dl3s85klfd").unwrap());
+    }
+
+    #[tokio::test]
+    pub async fn test_with_valid_user_address() {
+        let result = fetch_account_info_for_address(&MockClient::new(), &"erd1kj7l40rmklhp06treukh8c2merl2h78v2939wyxwc5000t25dl3s85klfd".into(), &CachingNone).await.unwrap();
+
+        assert_eq!(result.nonce, 6);
```

Comment:
Having one assert_eq! by field is error prone, because it's easy to miss a field. Can you derive PartialEq on AccountInfo and use one assert_eq! to compare the whole struct?